Scheduler

yannickl96 opened the pull request:
I also applied the TaskScheduler for the OPS-SAT repository. Furthermore, I implemented a ground profile for the SDK build.

dmarszk committed:
[platform-impl] Extend SDR and OPT-RX implementations

dmarszk committed:
[platform-impl] Improve ADCS, OPTRX, PowerCtrl, SDR

- Fix implementation bugs
- Improve handling of initialization errors

Commit:
Bug #321: Got rid of timer.scheduleAtFixedrate

TaskScheduler is used to replace timer.scheduleAtFixedRate calls.

Commit:
Bug #321: Got rid of timer.scheduleAtFixedrate

TaskScheduler is used to replace timer.scheduleAtFixedRate calls.

Commit:
Moved the content of the AntRun call to copy.xml.

In order to make it easier for the experimenter to build his apps in the right way I separated the Maven content from the Ant content.

Commit:
Moved the content of the AntRun call to copy.xml.

In order to make it easier for the experimenter to build his apps in the right way I separated the Maven content from the Ant content.

Commit:
Feature #360: Ground SDK profile for OPS-SAT.

I changed the SDK packaging so that the user can choose to build for the satellite or for ground simulation. By default, only the satellite components without any apps are built. The supervisor sim, ground-mo-proxy and apps can be built by providing the -Pground flag while invoking mvn install from the terminal.
Furthermore, I got rid of the timer.schedule.... calls in this repository as well.

Commit:
Feature #360: Ground SDK profile for OPS-SAT.

I changed the SDK packaging so that the user can choose to build for the satellite or for ground simulation. By default, only the satellite components without any apps are built. The supervisor sim, ground-mo-proxy and apps can be built by providing the -Pground flag while invoking mvn install from the terminal.
Furthermore, I got rid of the timer.schedule.... calls in this repository as well.

Commit:
Feature #360: Updated documentation.

Added documentation on the new SDK building process.

Commit:
Feature #360: Updated documentation.

Added documentation on the new SDK building process.

Commit:
Feature #415: Implemented setDesiredAttitudeMode.

It is now possible to activate every iADCS pointing mode provided by the iADCS API. Deactivating the modes is still WIP.

Commit:
Feature #415: Implemented setDesiredAttitudeMode.

It is now possible to activate every iADCS pointing mode provided by the iADCS API. Deactivating the modes is still WIP.

Commit:
Feature #415: Implemented unset method for iADCS.

The current attitude mode can now be unset by calling the unset() method in the adapter.

Commit:
Feature #415: Implemented unset method for iADCS.

The current attitude mode can now be unset by calling the unset() method in the adapter.

dmarszk commented on platform-service-impl/src/main/java/esa/mo/platform/impl/provider/opssat/AutonomousADCSOPSSATAdapter.java:
```diff
@@ -61,12 +74,102 @@ public AutonomousADCSOPSSATAdapter()
+    SEPP_IADCS_API_ORBIT_TLE_DATA tle = new SEPP_IADCS_API_ORBIT_TLE_DATA();
+    byte[] l1 = lines.get(0).getBytes();
+    byte[] l2 = lines.get(1).getBytes();
+    Logger.getLogger(AutonomousADCSOPSSATAdapter.class.getName()).log(Level.INFO,
```
Replace Logger.getLogger(~~) with static LOGGER

dmarszk commented on platform-service-impl/src/main/java/esa/mo/platform/impl/provider/opssat/AutonomousADCSOPSSATAdapter.java:
```diff
@@ -61,12 +74,102 @@ public AutonomousADCSOPSSATAdapter()
+    tolerance.setPREALIGNMENT_ANGLE_TOLERANCE_RAD(0.0872665f);
+    tolerance.setPREALIGNMENT_ANGLE_TOLERANCE_PERCENT(20.0f);
+    tolerance.setPREALIGNMENT_ANGULAR_VELOCITY_TOLERANCE_RADPS(0.00872665f);
+    tolerance.setPREALIGNMENT_TARGET_THRESHOLD_RAD(0.261799f); // See section 6.2.2.4 in ICD
```
Refactor magic constants into static final variables

Commit:
Feature #415: Fix method calls to service methods

Change the names of called methods due to changing
API description.

Commit:
Feature #415: Fix method calls to service methods

Change the names of called methods due to changing
API description.

Commit:
PR #1: Update README

Be more precise about the full use of the ground
SDK profile and be more coherent with the terminology.

Commit:
PR #1: Use correct logger

Use the static LOGGER to shorten the code. Also,
remove the comment inside the package POM.

Commit:
PR #1: Use correct logger

Use the static LOGGER to shorten the code. Also,
remove the comment inside the package POM.

Commit:
PR #1: Refactor magic constants

Put the tolerance constants as private static
final floats at the beginning of the class.

Commit:
PR #1: Refactor magic constants

Put the tolerance constants as private static
final floats at the beginning of the class.
